### PR TITLE
Fixes #6 - procfs path now is configurable

### LINKED ISSUE #6: Path to procfs is not configurable
The path to procfs is currently [hardcoded](https://github.com/intelsdi-x/snap-plugin-collector-disk/blob/master/disk/disk.go#L61-L67) to /proc. This makes it hard (if even possible) to use the plugin when snapd is run in a Docker container.

### PR DESCRIPTION
**Pull request by marcin-krolik**
- procfs path is no longer hardcoded, but defaults to "/proc"
- updated doc

### CODE REVIEW

File: disk/disk.go [hunk 1]
```diff
@@ -375,3 +390,27 @@ func parseMetricKey(k string) (disk, metric string) {
 	}
 	return result[0], result[1]
 }
+
+func resolveSrcFile(cfg interface{}) (string, error) {
+	// first configuration
+	if srcFile, err := config.GetConfigItem(cfg, "procfs_path"); err == nil {
+		fh, err := os.Open(srcFile.(string))
```
Comment (IzabellaRaulin) [inline disk/disk.go]: I made such like test that in task manifest config section was added with wrong procfs_path (intentionally) as below:

```
 "config": {
              "/intel/procfs/disk": {
                  "procfs_path": "/wrong_path/var/proc/diskstats"
                }
            },
```

There was no problem to run such task even I expected that error would occure, also no logs which would inform that default procfs_path was taken.


May return error should be considered in such case.

File: README.md [hunk 2]
```diff
@@ -183,6 +186,9 @@ Create a task JSON file (exemplary files in [examples/tasks/] (https://github.co
                 "/intel/procfs/disk/sdb2/pending_ops": {}								
             },
             "config": {
+              "/intel/procfs/disk": {
```
Comment (IzabellaRaulin) [inline README.md]: Updating config section with that change to existing exemplary task manifest would have been appreciated
(and +1 to keeping documentation consistency)

File: disk/disk.go [hunk 3]
```diff
@@ -160,36 +185,26 @@ func (dc *DiskCollector) CollectMetrics(mts []plugin.MetricType) ([]plugin.Metri
...
-
-		if err2 != nil {
-			return fmt.Errorf("Error open /proc/{diskstats|partitions}, errors = %s; %s", err, err2)
-		}
-
-		/* Kernel is 2.4.* */
-		fieldshift = 1
-		scanner = bufio.NewScanner(fh2)
+	if err != nil {
+		return fmt.Errorf("Error open /proc/{diskstats|partitions}, error = %s", err)
```
Comment (tiffanyfj) [inline disk/disk.go]: "Error opening...

File: disk/disk.go [hunk 4]
```diff
@@ -375,3 +390,29 @@ func parseMetricKey(k string) (disk, metric string) {
 	}
 	return result[0], result[1]
 }
+
+func resolveSrcFile(cfg interface{}) (string, error) {
+	// first configuration
+	if srcFile, err := config.GetConfigItem(cfg, "procfs_path"); err == nil {
```
Comment (tiffanyfj) [inline disk/disk.go]: Can also do 

`if srcFile, ok := config.GetConfigItem(cfg, "procfs_path"); ok {`

and later
`if !ok {`

**Comment by tiffanyfj** [comment]
Should rebase WIP into another commit or at least give it an actual message. Other than my one comment, this seems to be good to me.

**Comment by marcin-krolik** [comment]
@tiffanyfj Sure, I have not noticed that, thanks!

**Pull request merged**